pkg/engine/variables/vars_test.go

```diff
@@ -134,6 +134,149 @@ func Test_subVars_failed(t *testing.T) {
 	}
 }
 
+func Test_subVars_with_JMESPath_At(t *testing.T) {
+	patternMap := []byte(`
+	{
+		"kind": "{{@}}",
+		"data": {
+			"rules": [
+				{
+					"apiGroups": [
+						"{{request.object.metadata.name}}"
+					],
+					"resources": [
+						"namespaces"
+					],
+					"verbs": [
+						"*"
+					],
+					"resourceNames": [
+						"{{request.object.metadata.name}}"
+					]
+				}
+			]
+		}
+	}
+	`)
+
+	resourceRaw := []byte(`
+	{
+		"kind": "foo",
+		"name": "bar",
+		"metadata": {
+			"name": "temp",
+			"namespace": "n1"
+		},
+		"spec": {
+			"namespace": "n1",
+			"name": "temp1"
+		}
+	}
+		`)
+
+	expected := []byte(`{"data":{"rules":[{"apiGroups":["temp"],"resourceNames":["temp"],"resources":["namespaces"],"verbs":["*"]}]},"kind":"request.object.kind"}`)
```

Comment:
Should `kind` here be `foo` or `{{request.object.kind}}`?

Reply:
Yes. Fixed it.